### ISSUE TITLE
Plugin stuck in bad state. Target of URI doesn't exist 

### ISSUE THREAD

**Issue by johanlantz** [Dart-Code/Dart-Code#2953]
**Describe the bug**
I think this behaviour is due to the plugin but if it's someone else's responsibility please point me in the right direction.

I have a library project with a testapp in the example folder that has been working for a long time but out of the blue today the imports in the example app (that references files in the same testapp package) starts giving this error: Target of URI doesn't exist: 

This only affects the imports for files in the same example app, other imports coming from the library works as well as all other imports from other plugins. Something like:
```
import ‘package:flutter/material.dart’;
import ‘package:flutter/services.dart’;
import ‘package:rxdart/rxdart.dart’;
import ‘package:my_library/my_file.dart’;
import ‘package:my_library_example/demos/demo_widgets.dart’;  
````

The last import is from the testapp and that one is the one showing:
```
Target of URI doesn't exist: 'package:my_library_example/demos/demo_widgets.dart'.
Try creating the file referenced by the URI, or Try using a URI for a file that does exist.d
```

If I remove the package and use the relative path it works such as :
````
import ‘demos/demo_widgets.dart’;  
````

If I open the example app folder directly in VS Code, i.e I do not open the library project containing the example app, then it also works (I saw an old SO post about this but that it should be solved).

Other apps using the same folder structure works on the same machine.

The same project works fine on another computer opening the root project folder as I used to do up until now.

The project compiles fine from the terminal so it is the highlighting/analyze in VS Code that is broken, not the code itself.

I have tried:
- flutter packages get
- cache repair
- Uninstall flutter and dart VS code plugins
- restart the dart analyzis server in VS code

But nothing makes a difference. 

**To Reproduce**
I do not know what provoked this to start happening. The only unusual thing I did today was to checkout an identical copy of this project in another folder and then opened that folder in another VS Code window. I have no idea if its related.

**Expected behaviour**
The imports should work.

My only guess is that somewhere something has ended up in a weird state and maybe it's enough to just delete something but I do not know where to look :-)

**Versions (please complete the following information):**
 - VS Code version: 1.50.1
 - Dart extension version: 3.16.0
 - 1.22.3

**Comment by DanTup** [comment]
The package resolution is done using information in the `.packages` file or the newer `.dart_tool/package_config.json` file. You could check whether those files contain an entry for `my_library_example` that has the correct path to this project. If it's missing, or the path is wrong (for example if you renamed the project folder) then running the **Get Packages** command *should* fix it.

If not, is the project something you can share?

**Comment by johanlantz** [comment]
@DanTup Thank you so much for the quick reply. I can't share the full project and I am not sure it would help since on my other computer there is no issue (and all was fine at lunchtime today on this one as well). Getting a bit nuts here. Fortunately using the relative path in the imports hides the problem for now so I am not completely stuck.

I have checked both files indicated for the example app. The .package file has this info:
```
my_library:../lib/
my_library__example:lib/
```

and the package_config.json has:
```
{
      "name": "my_library_example",
      "rootUri": "../",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    }
```

These two files are identical on the working and non-working MacBooks. Both projects are cloned from the same master branch in the repo. I can also see in the timestamp that the package_config.json file on the non-working computer is regenerated when I resave the pubspec.yaml file.

So, as far as I can see, those things look ok but the imports for the example app are still marked with red in VS Code on the problematic computer. Happy to try anything, no matter how far out.

**Comment by DanTup** [comment]
In your `.packages` file you've written `my_library__example`. There appear to be two underscores in the second part, but that doesn't match the JSON - is this a typo, or is it like that?

Are the SDK versions on both machines the same?

Although they're cloned from the same repo, if you have files ignored in git, it's possible there are still differences between them. You could try copying the folder from one Mac to the other, or deleting all the ignored stuff (including the `.dart_tool` and `.packages` files at both levels, the app and the example) and re-fetching packages.

**Comment by johanlantz** [comment]
Hi again @DanTup. Sorry that was a typo, I just replaced the original project name.

So, it has started to work again and I have no explanation of why. I worked for a few hours with the relative imports instead for the testapp package and then before push the updates I put back the full imports using the package name and bam, problem gone. I have restarted VS Code as well and it stays happy.

I can not say what changed, at some point during working with the project something has unblocked itself. I did add a couple of new files in the library project but in the example app there are no new files, just some minor code changes.

So, I will wait until EOB friday and then I will close this ticket unless I come up with something more.

Thanks again for your help so far.
